schema/dsl/parse.go

```diff
@@ -341,36 +350,53 @@ func (p *parser) typeStruct() (*dmt.TypeDefnStruct, error) {
 			return nil, err
 		}
 		if tok == "(" {
+			details := dmt.StructRepresentation_Map_FieldDetails{}
 			p.consumePeeked()
-			if err := p.consumeRequired("implicit"); err != nil {
-				return nil, err
-			}
-
-			scalar, err := p.consumeToken()
-			if err != nil {
-				return nil, err
-			}
-			var anyScalar dmt.AnyScalar
-			switch {
-			case scalar[0] == '"': // string
-				s, err := strconv.Unquote(scalar)
+		parenLoop:
+			for {
+				tok, err = p.consumeToken()
 				if err != nil {
-					return nil, p.forwardError(err)
+					return nil, err
 				}
-				anyScalar.String = &s
-			case scalar == "true", scalar == "false": // bool
-				t := scalar == "true"
-				anyScalar.Bool = &t
-			default:
-				panic("TODO: implicit: " + scalar)
-			}
+				switch tok {
+				case ")":
```

Comment:
I'm choosing to only add abstraction layers as necessary in this code, e.g. when sharing code via a function pays for itself in code deduplication. Like you say, I just quite haven't gotten there for "paren lists", because I really have only implemented one such list implementation. The others are just one element.

> * I think I expected a comma between key-value pairs here, if there's more than one of them?  I know we found some other fixtures that don't have it, but those surprised me.  I think I'd advocate for adding it now and cleaning up things that are missing it as we find them.

I'd agree with adding commas. Definitely seems like an incremental change that we should do separately, though.

Reply:
Fair nuff.